src/resolve/resolved-dns-query.c

```diff
@@ -1377,3 +1382,12 @@ bool dns_query_fully_authoritative(DnsQuery *q) {
          * explicitly check previous redirects here.) */
         return (q->answer_query_flags & SD_RESOLVED_FROM_MASK & ~(SD_RESOLVED_FROM_TRUST_ANCHOR | SD_RESOLVED_FROM_ZONE)) == 0;
 }
+
+bool dns_question_has_record_types(DnsQuestion *q, Set *types) {
+        assert(q);
+        DnsResourceKey *key;
+
+        DNS_QUESTION_FOREACH(key, q)
+                return (set_contains(types, INT_TO_PTR(key->type)));
```

Comment:
the outer () are redundant.

with this you are now only checking the first entry...